src/unix/apple/disk.rs

```diff
@@ -431,3 +435,31 @@ unsafe fn new_disk(
         },
     })
 }
+
+
+/// Calls the provided closure in the context of a new autorelease pool that is drained
+/// before returning.
+/// 
+/// ## SAFETY:
+/// You must not return an Objective-C object that is autoreleased from this function since it
+/// will be freed before usable.
+unsafe fn with_autorelease<T, F: FnOnce() -> T>(call: F) -> T {
+    struct DrainPool {
```

Comment:
What's the point of having this struct? Couldn't both ffi functions be called before and after `call()`?

Reply:
This to to avoid any unnecessary leaking if `get_list` (or any other function in the future) panics and unwinds the stack. Without it the `pop` function wouldn't be called in these cases, leading to leaks. With this, the destructor will be called in the process. A lot of programs/libraries wrap calls in `std::panic::catch_unwind` so a panic here is not guaranteed to terminate the thread (and therefore free its memory) without something like this.

Reply:
Hum... I wonder here if it isn't too much considering this is `sysinfo` code only... Well, I'll let it up to you. If you want to stick with this version, please add a comment on the struct explaining why we do it this way please.

Reply:
I added the comment to provide context to future readers of this code.